packages/data-table-server/src/dataTable/createDataTable.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { TupaiaApiClient } from '@tupaia/api-client';
+import { DataTableType } from '../models';
+import { internalDataTables } from './internal';
+
+/**
+ * Factory function for building the correct data-table for the requested dataTableObject
+ */
+export const createDataTable = (dataTableObject: DataTableType, apiClient: TupaiaApiClient) => {
```

Comment:
We have two concepts with the same name here. I see you've tried to differentiate by calling one `dataTableObject` with the Type suffix on the ts type. 

However, I think more consistent with the rest of the codebase would be to call this first concept a dataTable (instead of dataTableObject) with DataTable (instead of DataTableType), and the other concept something else. 

It seems pretty similar to data broker, where you can have one DhisService which can pull data for many DhisInstance objects. So I'd suggest calling the second concept here a service as well, e.g. AnalyticsDataTableService. Thoughts?

Reply:
Ah great call! I love it, much tidier. 👍